cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/batch/AppWithTxAware.java

```diff
@@ -62,7 +62,7 @@ public void beforeSubmit(MapReduceContext context) throws Exception {
       job.setNumReduceTasks(0);
       FileOutputFormat.setOutputPath(job, new Path(context.getRuntimeArguments().get("outputPath")));
       PedanticTxAware input = context.getDataset("pedanticTxAware", ImmutableMap.of("value", "1"));
-      context.setInput("pedanticTxAware", input);
+      context.setInput("pedanticTxAware", ImmutableMap.of("value", "1"));
```

Comment:
Do we need to specify the properties twice? Once while getting the dataset and again while setting the input of the MapReduce job?